src/txmempool.h

```diff
@@ -35,6 +35,21 @@ inline bool AllowFree(double dPriority)
 /** Fake height value used in CCoins to signify they are only in the memory pool (since 0.8) */
 static const unsigned int MEMPOOL_HEIGHT = 0x7FFFFFFF;
 
+struct LockPoints
+{
+    // Will be set to the blockchain height and median time past
+    // values that would be necessary to satisfy all relative locktime
+    // constraints (BIP68) of this tx given our view of block chain history
+    int height;
+    int64_t time;
+    // As long as the current chain descends from the highest height block
+    // containing one of the inputs used in the calculation, then the cached
+    // values are still valid even after a reorg.
+    uint256 maxInputHash;
```

Comment:
hmm...  i feel like we did consider that, but i can't see now why that wouldn't work.  maybe it didn't work with some previous iteration.   it makes sense to switch to that if we can't see any problems with it.